### PR TITLE
Support qwen2-7b with fused decoderlayer optimization on NPU

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Add qwen2-7B NPU support with fused decoderlayer and multi process optimization.
Details: https://github.com/analytics-zoo/nano/issues/1576#issuecomment-2314969138

Different from other model support, we use QuantizedLinear instead of FusedQwenLowBitDecoderlayer during prefill process.

### 2. User API changes

N/A

### 4. How to test?
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [x] Application test

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py [hunk 1]
```diff
@@ -57,8 +57,8 @@ def optimize_llm(
         from transformers.models.llama.modeling_llama import LlamaForCausalLM
         from ipex_llm.transformers.npu_models.llama_mp import llama2_casullm_forward
         convert_forward(model, LlamaForCausalLM, llama2_casullm_forward)
-    elif model.config.model_type == "qwen2" and model.config.intermediate_size == 8960:
-        # for qwen2-1.5B
+    elif model.config.model_type == "qwen2":
+        # for qwen2-1.5B and qwen2-7B
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py:61]: How about Qwen2-0.5B or 72B?

Comment (plusbang) [reply python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py]: > How about Qwen2-0.5B or 72B? Need to add check

Sure, have added related check : )

File: python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py [hunk 2]
```diff
@@ -229,9 +229,15 @@ def mlp(self, hidden_states):
             hidden_states, self.intermediate_size, self.hidden_size, bias=False, wt_dtype=self.dtype
         )  # type: ignore[attr-defined]
         mm1 = self.eltwise_mul(self.swish(mm1), mm2)  # type: ignore[attr-defined]
-        hidden_states = self.linear(
-            mm1, self.hidden_size, self.intermediate_size, bias=False, wt_dtype=self.dtype
-        )
+        if self.intermediate_size == 18944:
```
Comment (yangw1234) [inline python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py:232]: maybe you can just override `self.mlp` in the sub-class

File: python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py [hunk 3]
```diff
@@ -383,7 +383,7 @@ def update_cache(self, past_key_value, indexes):
             self.load_cache_async()
 
     def load_cache_async(self):
-        self.load_wt_fn(len(self.input_ops), self._mm, self.kv_cache_c_handle)
+        self.load_wt_fn(len(self.input_ops), self._mm, self.kv_cache_c_handle, verify_size=True)
```
Comment (yangw1234) [inline python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py:386]: Not sure how much runtime overhead of `verify_size`, if it is large, maybe we can disable it at runtime and only use it when debugging?

**Review by yangw1234 — APPROVED** [review]

File: python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py (hunk 3 above)
Comment (plusbang) [reply python/llm/src/ipex_llm/transformers/npu_models/mp_models_base.py]: > Not sure how much runtime overhead of `verify_size`, if it is large, maybe we can disable it at runtime and only use it when debugging?

According to my experiment, the overhead seems quite small. BTW have removed it : )

**Pull request merged**